### PR TITLE
Implements blog functionality for visitors

### LINKED ISSUE #65: Implement a blog functionality for Visitors
Add a "Blog" section to the app's navigation bar, providing users with easy access to the blog functionality. Enable registered users to create and publish their travel blogs, stories, and articles. Allow users to upload photos to accompany their blog posts.
Create a separate page accessible through the "Blog" section where all user-generated blogs are listed. Display blogs in a sorted order, with the most recent posts at the top.
When a user clicks on a blog entry in the list, it should expand to display the full content of the blog post, including text and accompanying photos. Implement "Like" and "Dislike" buttons for each blog post, allowing users to express their preference.
Include a comment section for each blog post, enabling users to engage in discussions and leave feedback.
Utilize the existing review/comment functionality from the "Places" section and adapt it for blog posts.
(Will give bonus for neat work 😋)

Note:

- Create a new branch for every task to isolate changes
- Start working on the issue only when it is assigned to you
- Submit the PR on time, no extra time will be allotted

### PR DESCRIPTION
**Pull request by Nandika-A**
## Description

A new django app 'blog' is created.

models:
1. post: All details related to post including foreign key to user, like and dislike models.
2. like: Stores all the likes for every posts by every visitor.
3. dislike: Stores all the dislikes for every posts by every visitor.
4. comment: Stores comments.

views:

Functionality for List View, Detail View, Create, Update, Delete, like, dislike, comment on posts, and view the user's personal posts.

1. The functionality can be accessed by clicking on the blog tab on the Navigation bar. This leads to the list view of all the blogs.
2.  By clicking on the post title, users can access the detailed view of that post, where they can also like, dislike and comment if they are authenticated.
3.  If the user is that post's author, they can update or delete the post by clicking on the buttons provided. 
4. If a visitor likes/dislikes the post, the count is incremented/decremented. 
5. Comments can be viewed and new comments can be added by clicking on comment button, which directs to the comment page. 
6. Every page contains a side-bar, providing options to create new post, view personal posts and view profile if the user is authenticated, otherwise mentions them to login/signup.
7. 'Your posts' tab directs the user to the list view of all their posts.

## Related Issue

Fixes #65 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Motivation and Context

Adds a blog feature for visitors.

## How Has This Been Tested?

There are no errors after the implementation.

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

**Comment by Nandika-A** [comment]
@kunal2812 I have implemented all the functionalities specified in the issue. Please check if any changes are required.

**Comment by kunal2812** [comment]
Remove auto-generated migration files and binary files in pycache folders and add those folders to .gitignore. Keep only those changes which are relevant to the issue resolved

**Comment by Nandika-A** [comment]
@kunal2812 Sir I have made the required changes.

File: blog/__pycache__/__init__.cpython-310.pyc
Comment (kunal2812) [inline blog/__pycache__/__init__.cpython-310.pyc:1]: Remove these as well

**Comment by kunal2812** [comment]
@Nandika-A The contents of the blog post are not visible when I click on the blog post.

image

Also please follow the website theme. Currently, it looks like a separate website altogether. Also when I comment on any post they should be visible below the post. As of now I can't see them anywhere.

Add the documentation changes as well.

```
python manage.py makemigrations blog
python manage.py migrate blog
```

**Comment by Nandika-A** [comment]
@kunal2812 Sir I have made a separate page for viewing comments. I will move to the same page now.

**Comment by Nandika-A** [comment]
@kunal2812 I have moved the comments to the post detail page and updated the readme.
Screenshot (53)
Screenshot (54)

**Comment by kunal2812** [comment]
@Nandika-A Fix the theme of this page. This issue has bonus points.

**Comment by Nandika-A** [comment]
@kunal2812 sir I included the website navbar in blog/base.html but it just included the HTML and not the CSS.

**Comment by kunal2812** [comment]
You can customise your existing navbar according to theme of the app

**Comment by kunal2812** [comment]
> @kunal2812 I have moved the comments to the post detail page and updated the readme. Screenshot (53) Screenshot (54)

Why is it showing "None" in the comment section. It should show the username of the person who commented right ?

**Comment by Nandika-A** [comment]
> > @kunal2812 I have moved the comments to the post detail page and updated the readme. Screenshot (53) Screenshot (54)
> 
> Why is it showing "None" in the comment section. It should show the username of the person who commented right ?

sir in visitor/models, the username is set to none.

**Comment by Nandika-A** [comment]
Should I use email instead of username here?

**Comment by kunal2812** [comment]
Yes just like the blogs

**Comment by Nandika-A** [comment]
@kunal2812 Sir I have changed the theme.

**Comment by Nandika-A** [comment]
@kunal2812 sir, are there any changes to be made?

**Comment by kunal2812** [comment]
@Nandika-A Ensure that the image is properly contained within the blog body, preventing it from flowing all over the page

image

**Comment by Nandika-A** [comment]
@kunal2812 sir I have fixed the image size now.

**Comment by kunal2812** [comment]
image

Login/Signup button isn't working

image

Timing is in UTC. Change to local time zone (The time was 6:50 pm when I took the screenshot).

Fix these issues. Rest everything is okay.

File: media/images/post/Screenshot_21.png
Comment (kunal2812) [inline media/images/post/Screenshot_21.png:1]: Is this file needed ?

File: media/images/Visitors/cat_discord.png
Comment (kunal2812) [inline media/images/Visitors/cat_discord.png:1]: This one ?

File: media/images/post/cat_discord_Qw7xVwD.png
Comment (kunal2812) [inline media/images/post/cat_discord_Qw7xVwD.png:1]: Remove unnecessary files.

**Review by kunal2812 — CHANGES_REQUESTED** [review]

**Comment by Nandika-A** [comment]
I will delete these files

**Comment by Nandika-A** [comment]
> image
> 
> Login/Signup button isn't working
> 
> image
> 
> Timing is in UTC. Change to local time zone (The time was 6:50 pm when I took the screenshot).
> 
> Fix these issues. Rest everything is okay.

Sir, I have directly imported the navbar from navbar.html. I haven't changed anything in the code of login/signup.

**Comment by Nandika-A** [comment]
@kunal2812 sir I have made the changes.

**Review by kunal2812 — APPROVED** [review]

**Pull request merged**